### DISCUSSION TITLE
CO2 value and unit with icon too long

### DISCUSSION THREAD

**Discussion by quenthal** [Blackymas/NSPanel_HA_Blueprint#858] [Q&A] [answered]
Couldn't find the discussion, even though I think I have seen it somewhere.

Id like to put CO2 value at the upper right corner. However the value is usually around 400.0 ppm with very bad quality over 1000.0 ppm. With icon that string is too long. Is there any way to make the font smaller, lose the .x precision, lose the unit string (ppm) or move the text little bit left? Templating another co2 sensor might be partial solution.

**Comment by edwardtfn** [comment]
Which screen are you talking about?

**Reply by quenthal** [reply]
Main screen, sensor slot upper right screen. Also the icon is slightly cropped at the bottom.

**Reply by rleongcs** [reply, marked as answer]
@edwardtfn I believed @quenthal is refering to "Sensor 03" on the Home page. I did call out the icon issue in #692. As for the decimal point, not sure why HA is passing that along even though we did set the precision of the entity to not have decimal as discussed in [#546](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/546#issuecomment-1520609070).

**Reply by edwardtfn** [reply]
The support for precision was introduced on HA v2023.3, but we are still keeping support for v2022.10, so we cannot take advantage of that. Also, that is not available for attributes, and we use it in many different places.
But maybe we should change minimum requirements to 2023.3 and add this support in some of those entities...
I will try to find some time to look at this.

**Reply by rleongcs** [reply]
I just updated to the latest v3.5_dev and confirmed it is working.

![IMG_1817](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/61628312/ba749bf1-3021-400b-83b9-3edfe4a3cdaa)

**Reply by edwardtfn** [reply]
It's good to hear that. With #880 we started using the same precision set in Home Assistant, when available.
We still have some values that are coming from attributes, where precision cannot be set. In these cases we are using the same precision as the previous versions.
